src/ocl_kernel.cpp

```diff
@@ -61,6 +64,12 @@ void OCLKernelInvoke::run() const
 
     MIOPEN_HANDLE_LOCK
 
+    const char* const arch = miopen::GetStringEnv(MIOPEN_DEVICE_ARCH{});
+    if(arch != nullptr && strlen(arch) > 0)
+    {
+        MIOPEN_THROW("MIOPEN_DEVICE_ARCH used, escaping launching kernel");
```

Comment:
Cant we just return quietly and issue a log instead ?

Reply:
we need to throw otherwise it will try and load